src/drake/fs.clj

```diff
@@ -16,6 +16,19 @@
 (def drake-ignore "Names of files or directories to be ignored by Drake"
   #{"_logs"})
 
+(defn get-directory
+  [path]
+  (loop [f (fs/file path)]
+    (cond
+     (.isDirectory f) f
+     (nil? f) nil
+     :else (recur (.getParentFile f)))))
+
```

Comment:
I'm objecting to get-directory-path, not get-directory. It's fine to look up a thing's directory, but returning it as a File is the right thing to do. I'd rather defer the call to `.getPath` to when we call `shell`, so that we're using File objects wherever possible, and then strings only at the outermost edges where we have to interact with something that needs a string.